### PR TITLE
(SERVER-119) Properly sign puppet extension values

### PR DESCRIPTION
**Pull request by dankreek**

### CODE REVIEW

**Comment by jeffmccune** [comment]
It'd be great if the commit message for 4943604 contained at least the following information:  The behavior of the system without the patch applied, why this is a problem, and how the change resolves the problem.

File: src/java/com/puppetlabs/ssl_utils/SSLUtils.java [hunk 1]
```diff
@@ -189,28 +196,43 @@ public static X509Certificate signCertificate(String issuerDn,
                                                   String subjectDn,
                                                   PublicKey subjectPublicKey,
                                                   List<Map<String, Object>> extensions)
-            throws IOException, OperatorCreationException, CertificateException, NoSuchAlgorithmException, SignatureException, InvalidKeyException {
-        X509V3CertificateGenerator certGen = new X509V3CertificateGenerator();
```
Comment (nwolfe) [inline src/java/com/puppetlabs/ssl_utils/SSLUtils.java]: Oh sweet, so we were able to get rid of using this deprecated class? Finally. Looks like we can get rid of the import statement for this class (surprised IDEA didn't take care of that).

**Comment by nwolfe** [comment]
Testing hipchat webhook...

**Comment by fpringvaldsen** [comment]
I'm :+1: on this barring @jeffmccune's comment about the commit message. My knowledge of this is limited, however, so someone else should definitely take a look at it.

**Comment by jeffmccune** [comment]
I've got no other comments either, I'm :+1: except the housekeeping comment.

Looks good.

**Comment by dankreek** [comment]
If ya'lls ready to merge, let me know and I can amend that commit message before merge

**Pull request merged**